### PR TITLE
Make included tests pass

### PR DESCRIPTION
**Pull request by rsocci**

### CODE REVIEW

File: test/json_api/assert_payload_test.exs [hunk 1]
```diff
@@ -909,4 +909,42 @@ defmodule Voorhees.Test.JSONApi.AssertPayloadTest do
       })
     end
   end
+
+  test "should pass with equal payloads containing both data and included" do
```
Comment (danmcclain) [inline test/json_api/assert_payload_test.exs:913]: I'd like a failing test with a single failure in both data and included, data message coming before included

File: lib/voorhees/json_api.ex [hunk 2]
```diff
@@ -55,17 +55,41 @@ defmodule Voorhees.JSONApi do
...
+
+  defp merge_results(existing, new) do
+    case {existing, new} do
+      {:ok, _} -> new
+      {_, :ok} -> existing
+      _ ->
+        {:error, existing_message} = existing
+        {:error, new_message} = new
+        {:error, elem(existing, 1) <> elem(new, 1)}
+    end
```
Comment (danmcclain) [inline lib/voorhees/json_api.ex:83]: I rather use pattern matching than elem. I would even add the case right in `compare_payloads`, as it is not going to be anywhere else but there, saving a hop

```
case {existing, new} do
  {:ok, state} -> state
  {state, :ok} -> state
  {{:error, existing_message}, {:error, new_message}} ->
    {:error, existing_message <> new_message}
end
```

File: lib/voorhees/json_api.ex [hunk 3]
```diff
@@ -55,17 +55,41 @@ defmodule Voorhees.JSONApi do
   defp error_message(:ok), do: ""
   defp error_message({:error, message}), do: "Payload did not match expected\n\n" <> message
 
+  defp compare_property(actual, expected, property_name, options) do
+    actual_value = Map.fetch(actual, property_name)
+    expected_value =  Map.fetch(expected, property_name)
+
+    case {actual_value, expected_value} do
+      {:error, :error} -> :ok
+      {:error, _} -> compare_resources(actual_value, expected_value, options)
```
Comment (danmcclain) [inline lib/voorhees/json_api.ex]: Again, use pattern matching here, then you can put the fetch in the case statement on 62

File: lib/voorhees/json_api.ex [hunk 4]
```diff
@@ -55,17 +55,41 @@ defmodule Voorhees.JSONApi do
...
+      {_, :error} -> nil
+      {_, _} ->
+         compare_resources(elem(actual_value, 1), elem(expected_value, 1), options)
```
Comment (danmcclain) [inline lib/voorhees/json_api.ex]: Pattern matching removes need for elem

File: lib/voorhees/json_api.ex [hunk 5]
```diff
@@ -55,17 +55,39 @@ defmodule Voorhees.JSONApi do
...
   defp error_message({:error, message}), do: "Payload did not match expected\n\n" <> message
 
+  defp compare_property(actual, expected, property_name, options) do
+    actual_value = Map.fetch(actual, property_name)
+    expected_value =  Map.fetch(expected, property_name)
+
+    case {Map.fetch(actual, property_name), Map.fetch(expected, property_name)} do
+      {:error, :error} -> :ok
+      {:error, _} -> compare_resources(actual_value, expected_value, options) # need to revisit & test
+      {_, :error} -> nil # need to revisit
```
Comment (rsocci) [inline lib/voorhees/json_api.ex]: Still need to come back to this

File: test/json_api/assert_payload_test.exs [hunk 6]
```diff
@@ -909,4 +909,164 @@ defmodule Voorhees.Test.JSONApi.AssertPayloadTest do
...
+    }
+  end
+
+  test "throw an error when payload does not contain data or included, but is expected" do
+    payload = %{}
+
+    assert_raise ExUnit.AssertionError, """
+    Payload did not match expected
+
+    "data" did not match expected
```
Comment (danmcclain) [inline test/json_api/assert_payload_test.exs]: Would rather have something that says `"data" was expected, but was not present`

**Pull request merged**